src/lxc/tools/lxc_start.c

```diff
@@ -284,27 +278,8 @@ int main(int argc, char *argv[])
 		}
 	}
 
-	for (i = 0; i < LXC_NS_MAX; i++) {
-		const char *key, *value;
-
-		value = my_args.share_ns[i];
-		if (!value)
-			continue;
-
-		if (i == LXC_NS_NET)
-			key = "lxc.namespace.net";
-		else if (i == LXC_NS_IPC)
-			key = "lxc.namespace.ipc";
-		else if (i == LXC_NS_UTS)
-			key = "lxc.namespace.uts";
-		else if (i == LXC_NS_PID)
-			key = "lxc.namespace.pid";
-		else
-			continue;
-
-		if (!c->set_config_item(c, key, value))
-			goto out;
-	}
+	if (!lxc_ns_setup(&my_args, c))
```

Comment:
how about lxc_ns_sharing_setup() to be more specific?

Reply:
`lxc_setup_shared_ns()` (just to throw one more into the mix :))

Reply:
yeah I like that better.

Reply:
Sounds good, I'll rename it.